### DISCUSSION TITLE
v3.0.7

### DISCUSSION THREAD

**Discussion by alexec** [argoproj/argo-workflows#6010] [General]
# Quick Start

## What's New?

Find out in [our blog post](https://blog.argoproj.io/argo-workflows-v3-0-4d0b69f15a6e)

## Migration From v2

v3 is not intended to introduce significant breaking changes for most users. Please read the "Known issues" and "Breaking changes" to determine if you might be impacted.

## Known Issues

* Some users are reporting that the UI attempts to open all listing workflows in all namespaces, even if the user does not have access. To work-around - append the correct namespace in the URL.
* OpenShift does not appear to support the coordination API which is mandatory for leader election. #5638 #5306 
* Archiving of logs no longer works. #5639

### chore(server): Enable TLS by default. Resolves #5205 (#5212)

The server now starts with TLS enabled by default if a key is available.  This can be disabled using `--secure=false`.

If you have an ingress, you may need to add the appropriate annotations:(varies by ingress):

```
alb.ingress.kubernetes.io/backend-protocol: HTTPS
nginx.ingress.kubernetes.io/backend-protocol: HTTPS
```

### Mac

Available via `curl`

```sh
# Download the binary
curl -sLO https://github.com/argoproj/argo/releases/download/v3.0.7/argo-darwin-amd64.gz

# Unzip
gunzip argo-darwin-amd64.gz

# Make binary executable
chmod +x argo-darwin-amd64

# Move binary to path
mv ./argo-darwin-amd64 /usr/local/bin/argo

# Test installation
argo version
```

### Linux

Available via `curl`

```sh
# Download the binary
curl -sLO https://github.com/argoproj/argo/releases/download/v3.0.7/argo-linux-amd64.gz

# Unzip
gunzip argo-linux-amd64.gz

# Make binary executable
chmod +x argo-linux-amd64

# Move binary to path
mv ./argo-linux-amd64 /usr/local/bin/argo

# Test installation
argo version
```

## Argo Controller

```sh
kubectl create namespace argo
kubectl apply -n argo -f https://raw.githubusercontent.com/argoproj/argo-workflows/v3.0.7/manifests/install.yaml
```

### Since `v3.0.6`

- b6e986c85 fix(controller): Increase readiness timeout from 1s to 30s (#6007)

### Since `v3.0.5`

- 10ecb7e5b fix(controller): Listen on :6060 (#5988)

### Since `v3.0.4`

- f893ea682 feat(controller): Add liveness probe (#5875)
- e64607efa fix(controller): Empty global output param crashes (#5931)
- eeb5acba4 fix(ui): ensure that the artifacts property exists before inspecting it (#5977)
- 49979c2fa fix(controller): Revert cb9676e88857193b762b417f2c45b38e2e0967f9. Fixes #5852 (#5933)

### Since `v3.0.3`

-  v3: `podSpecPatch` can result in failed workflow
- Update manifests to v3.0.4\
- fix(ui): Fix workflow summary page unscrollable issue
- fix: Fix active pods count in node pending status with pod deleted.
- fix(controller): Fix pod spec jumbling. Fixes #5897

#### Contributors

* Alex Collins 
* Radolumbo 
* Saravanan Balasubramanian 
* dinever 

<hr /><em>This discussion was created from the release <a href='https://github.com/argoproj/argo-workflows/releases/tag/v3.0.7'>v3.0.7</a>.</em>

**Comment by KenGant** [comment]
We'd like to use Argo workflows on a project that requires our cyber security people to approve everything used.  We ran a scan with Nexus IQ that produced a number of vulnerability reports that have to be addressed for us to use it.  I'd love to be able to discuss this with someone in a position to act on that, and ... I'm not sure of the etiquette here.  This seemed like as good a place as any to start. I would have attached the report, but a) it was kinda big, and b) I didn't know if you wanted it, or wanted it here :-)

**Reply by agilgur5** [reply]
An issue would be the proper place for that. I would imagine most of those have been resolved. We do run [regular vuln scans with Snyk](https://argo-workflows.readthedocs.io/en/latest/static-code-analysis/) etc and Argo's images are all distroless as well, dramatically limiting the attack surface area.
